### PR TITLE
Add dev-only Reload button to sidebar + immediate page reload

### PR DESCRIPTION
**Pull request by fregante**
## What does this PR do?

- Follows https://github.com/pixiebrix/pixiebrix-extension/pull/7354

Two changes:

- add a (DEV-only) always-visible reload button to the sidebar, like the page editor
- when the sidebar shows the error boundary, have the reload happen in-place:
	- without going through the content scrip
	-  without closing/reopening animation

## Demo

This demo shows the dev-only button + the "fast reload" behavior

> [!IMPORTANT] 
> This specific reload button is developer-only, but it there's already one in the `ErrorBoundary` for everyone.

https://github.com/pixiebrix/pixiebrix-extension/assets/1402241/826bb5d6-29c9-4296-9ed1-d51b108c17ee

## Checklist

- [x] Designate a primary reviewer: @mnholtz  

## Slightly related

- https://github.com/pixiebrix/pixiebrix-extension/issues/7649

### CODE REVIEW

File: src/sidebar/Header.tsx [hunk 1]
```diff
@@ -19,19 +19,38 @@ import React from "react";
...
 const Header: React.FunctionComponent = () => {
   const { logo, showSidebarLogo, customSidebarLogo } = useTheme();
   const theme = useGetThemeName();
   /* In MV3, Chrome offers a native Close button */
   const showCloseButton = !isMV3();
 
+  const { flagOn } = useFlags();
+  const showDeveloperUI =
+    process.env.ENVIRONMENT === "development" ||
+    flagOn("page-editor-developer");
```
Comment (fregante) [inline src/sidebar/Header.tsx:47]: I reused the flag we have in the page editor. I can drop it if desired.

File: src/contentScript/messenger/strict/api.ts [hunk 2]
```diff
@@ -31,7 +31,6 @@ export const getReservedSidebarEntries = getMethod(
 );
 export const updateSidebar = getNotifier("UPDATE_SIDEBAR");
 export const sidebarWasLoaded = getNotifier("SIDEBAR_WAS_LOADED");
-export const reloadSidebar = getMethod("RELOAD_SIDEBAR");
```
Comment (fregante) [inline src/contentScript/messenger/strict/api.ts:34]: Fewer methods here, fewer methods to make strict:

- https://github.com/pixiebrix/pixiebrix-extension/issues/6526

File: src/sidebar/Header.tsx [hunk 3]
```diff
@@ -19,19 +19,38 @@ import React from "react";
...
+
+  const classes = cx(
```
Comment (mnholtz) [inline src/sidebar/Header.tsx:49]: nit: `classes` -> `buttonStyles`

File: src/sidebar/Header.tsx (hunk 1 above)
Comment (mnholtz) [reply src/sidebar/Header.tsx:47]: Is there a reason to put it under a feature flag vs. an env variable?

**Review by mnholtz — COMMENTED** [review]
I have some questions, but logic looks good so far

File: src/sidebar/Header.tsx (hunk 1 above)
Comment (fregante) [reply src/sidebar/Header.tsx:47]: I don't know, I just copied the one we already have. For me `process.env.ENVIRONMENT === "development"` would be enough

File: src/sidebar/Header.tsx [hunk 4]
```diff
@@ -62,15 +81,23 @@ const Header: React.FunctionComponent = () => {
...
         </div>
       )}
+      {showDeveloperUI && (
+        <Button
+          type="button"
+          size="sm"
+          variant="link"
+          onClick={reloadPage}
+          className={classes}
+        >
```
[the remark this answers is not in the document]
Comment (fregante) [reply src/sidebar/Header.tsx:92]: Done

**Review by grahamlangford — APPROVED** [review]

**Pull request merged**